### PR TITLE
[Feature store] Explode if using spark and target path is without schema

### PR DESCRIPTION
**Pull request by dinal**

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -476,6 +476,10 @@ def _ingest_with_spark(
         targets = [get_target_driver(target, featureset) for target in targets]
 
     for target in targets or []:
+        if target.path and urlparse(target.path).scheme == "":
```
Comment (urihoenig) [inline mlrun/feature_store/api.py:479]: This is post spark session creation, I guess that the whole block has to be inside try-finally with _post_ingestion (not only your code, but especially this exception).

File: mlrun/feature_store/api.py [hunk 2]
```diff
@@ -451,38 +451,53 @@ def _ingest_with_spark(
...
+            spark_options = target.get_spark_options(key_column, timestamp_key)
+            logger.info(
+                f"writing to target {target.name}, spark options {spark_options}"
+            )
+            df.write.mode("overwrite").save(**spark_options)
+            target.set_resource(featureset)
+            target.update_resource_status("ready")
+        _post_ingestion(mlrun_context, featureset, spark)
+    finally:
+        if spark:
```
Comment (urihoenig) [inline mlrun/feature_store/api.py:499]: _post_ingestion stops the spark session too, this means calling it twice in a successful flow.
I think that you should either call _post_ingestion with a success/error indication from finally or remove the spark.stop from _post_ingestion.

Comment (dinal) [reply mlrun/feature_store/api.py:499]: i'll remove spark.close() from _post_ingestion

**Review by Hedingber — APPROVED** [review]

File: mlrun/feature_store/api.py (hunk 2 above)
Comment (urihoenig) [reply mlrun/feature_store/api.py:499]: sorry for nagging but it seems that _post_ingestion is called from ingest as well (there's a chance that there won't be a spark context there but we have to be sure)

Comment (urihoenig) [reply mlrun/feature_store/api.py:499]: I see that we cannot get to this point with spark_context so please ignore

**Review by urihoenig — APPROVED** [review]

**Pull request merged**